doozerlib/dblib.py

```diff
@@ -1,64 +1,366 @@
+                            user=os.environ[constants.DB_USER],
+                            password=os.environ[constants.DB_PWD],
+                            database=os.environ[constants.DB_NAME] if constants.DB_NAME in os.environ
+                            else constants.default_db_params[constants.DB_NAME])
+                        break
+                    except:
+                        traceback.print_exc()
+                        db.runtime.logger.error("Unable to connect to mysql database.")
+                        time.sleep(25 * count)
+                        count += 1
+
+            func(*args, **kwargs)
+        else:
+            db.connection = None
+
+    return wrapper
+
+
 class DB(object):
 
     def __init__(self, runtime, environment, dry_run=False):
         """
         :param runtime: The runtime
         :param environment: int / stage / prod  for database environment. None will not write any data.
-        :param dry_run: If True, not data will be written
         """
-        self.lock = threading.Lock()
-        self.existing_domains = {}  # maps existing domain names to True
+
         self.runtime = runtime
         self.environment = environment
-        self.cmd_line = ','.join(sys.argv)
-        m = hashlib.md5()
-        m.update(self.cmd_line.encode('utf-8'))
-        self.uniq = m.hexdigest()  # help guarantee record uniqueness
+        self.dry_run = dry_run
 
-        self.client = None
-        if environment and not dry_run:
-            try:
-                self.client = boto3.client('sdb', region_name='us-east-1')
-                for name in self.client.list_domains()['DomainNames']:
-                    self.existing_domains[name] = True
-            except:
-                traceback.print_exc()
-                runtime.logger.error('Unable to acquire simpledb client; please check AWS creds')
-
-    def create_domain_if_missing(self, domain):
-        if self.client:
-            with self.lock:
-                if domain in self.existing_domains:
-                    # Already exists
-                    return
-                self.client.create_domain(
-                    DomainName=domain,
-                )
-                self.existing_domains[domain] = True
+        # mysql config for build records
+        self.mysql_db_env_var_setup = True
+        self.mysql_db_env_var_setup = self.check_missing_db_env_var() and self.check_database_exists()
+
+        self.lock = threading.Lock()
+
+        self.connection = None
+        self.column_identifier = self.create_column_identifier()
+
+        self._table_column_cache = {}
+
+    def check_missing_db_env_var(self):
+
+        """
+        The function checks if the expected environment variables are set. If not, raise an exception.
+        """
+
+        if not (constants.DB_USER in os.environ and constants.DB_PWD in os.environ and constants.DB_NAME in os.environ):
```

Comment:
Noted. Yes, it should pick the default. I had deliberately avoided default for DB_NAME initially since I was not sure which MySQL will we test this on. And to be safe just make it must parameter. I'll switch it back to the default value.